inherit from fieldcell and make nicer appearance in a list of cells

kradical opened the pull request:
Hello! Thank you for this library, it is the only custom eureka cell I use and is a great one.

I made this pull request to improve the row's behaviour in the context of a multi-row form. Inheriting from _FieldCell and KeyboardReturnHandler means the keyboard will have Next and Return on the keyboard with the expected "Eureka Standard" actions. It also gives the same TextFieldDelegate methods as all the other FieldCells so that the behaviour around text entry should be the same.

I also modified the constraints of the cell:
-changed the TextField's leading edge to 22 (from 10, aligns with other Eureka Text rows)
-changed the password strength indicators leading edge to 10 (instead of match the TextField).

Thanks again,
Konrad

kradical committed:
inherit from fieldcell and make nicer appearance in a list of cells

kradical commented:
@dernster Could you please take a look at this? I don't think I can add reviewers.

dernster commented:
@kradical I'll take a look asap. Thanks for making this PR :)

dernster commented:
@kradical Hey, reviewing this... All Eureka rows seems to have 16 of leading spacing in their textfield. I've tried your branch and it looks like this:

![screen shot 2017-06-21 at 4 59 22 pm](https://user-images.githubusercontent.com/5302420/27404045-036c7290-56a3-11e7-9cbb-43701e0aa0ba.png)

I think we need to default it to 16 and have that same padding with the password strength view. I would like to optionally change those numbers, so maybe having outlets for those constraints would be great. What do you think?

kradical commented:
Yeah I agree, I was just testing on a smaller device and was about to message you, the difference is very obvious on a small device! I like the idea of having outlets for those numbers, I could try implementing and update the PR sometime over the next week?

kradical committed:
add outlets for leading and trailing constraints on cell content

kradical commented:
I added outlets and changed the default leading and trailing space to 16 (matches the title and value in a TextRow). The following code is how the new constraints could be used:
```
...
            <<< GenericPasswordRow { row in
                row.tag = "New Password"
                row.placeholder = "New Password"
                row.passwordValidator = CustomPasswordValidator()
            }
            <<< GenericPasswordRow { row in
                row.tag = "Confirm Password"
                row.placeholder = "Confirm Password"
                row.passwordValidator = ConfirmPasswordValidator()
            }.cellSetup { cell, _ in
                cell.leading.constant = 50.0
                cell.trailing.constant = 50.0
            }
...
```

Which results in one standard sized password row and one with custom leading and trailing spacing.

![screen shot 2017-06-21 at 2 18 21 pm](https://user-images.githubusercontent.com/9980985/27407288-9b7c7a4c-568c-11e7-8e6f-86a8cd30d60b.png)

dernster commented:
@kradical That looks great! Does the password strength view has the same width as the textfield now? I think that should be the default at least.

kradical commented:
the password strength view is tied to the textfield on the leading edge and the visibility icon on the trailing edge! and cell.trailing is the trailing constraint on the visibility icon.
